tsdb/index/tsi1/index.go

```diff
@@ -826,6 +879,13 @@ func (i *Index) TagSets(name []byte, opt query.IteratorOptions) ([]*query.TagSet
 	return sortedTagsSets, nil
 }
 
+// DiskSizeBytes returns the size of the index on disk.
+func (i *Index) DiskSizeBytes() int64 {
+	fs := i.RetainFileSet()
+	defer fs.Release()
+	return fs.Size()
```

Comment:
`Shard.DiskSize` is called fairly frequently.  It created a lot of lock contention in tsm and I suspect it will cause a problem here as well.  I ended up switching the sizes in tsm to be atomic counters to fix it.

Reply:
@jwilder this calls down to [here](https://github.com/influxdata/influxdb/blob/master/tsdb/index/tsi1/index_file.go#L122) on each file in the fileset. It's should be cheap.